backend/app/infrastructure/routes/rewards.py

```diff
@@ -413,3 +428,21 @@ def get(self, epoch: int):
         app.logger.debug(f"Merkle tree leaves for epoch {epoch}: {merkle_tree_leaves}")
 
         return merkle_tree_leaves.to_dict()
+
+
+@ns.route("/current_user_budget")
+@ns.doc(description="Returns current user budget based on if allocation happened now.")
+class CurrentUserBudget(OctantResource):
+    @ns.expect(current_user_budget_model_request)
+    @ns.marshal_with(current_user_budget_response)
+    @ns.response(200, "Current user budget successfully retrieved")
```

Comment:
`current` might be a bit misleading, I'd suggest to replace it with `estimated` in this flow

Reply:
or `upcoming` or `upcoming estimated`, sth more precise then `current` ;)